packages/cloud-functions/src/circuitBreaker/circuitBreaker.ts

```diff
@@ -0,0 +1,89 @@
+import * as functions from 'firebase-functions'
+import { KeyManagementServiceClient, protos } from '@google-cloud/kms'
+import express from 'express'
+import { CIRCUIT_BREAKER_KEY_PATH } from '../config'
+
+type CryptoKeyVersionState = keyof typeof protos.google.cloud.kms.v1.CryptoKeyVersion.CryptoKeyVersionState
+
+const client = new KeyManagementServiceClient()
+
+const app = express()
+app.use(express.json())
+
+async function getKeyState(): Promise<CryptoKeyVersionState> {
+  const getCryptoKeyVersionResult = await client.getCryptoKeyVersion({
+    name: CIRCUIT_BREAKER_KEY_PATH,
+  })
+  return getCryptoKeyVersionResult[0].state as CryptoKeyVersionState
+}
+
+app.get('/health', async (req, res) => {
+  return res.status(200).json({
+    status: 'ok',
+  })
+})
+
+app.get('/status', async (req, res) => {
+  const keyState = await getKeyState()
+  switch (keyState) {
+    case 'ENABLED':
+      return res.status(200).json({
+        status: 'enabled',
+      })
+    case 'DISABLED':
+      return res.status(200).json({
+        status: 'temporarily disabled',
+      })
+    case 'DESTROYED':
+      return res.status(200).json({
+        status: 'permanently disabled',
+      })
+    default:
+      console.error(`Unexpected key state: ${keyState}`)
+      return res.status(200).json({
+        status: 'unknown',
+      })
+  }
+})
+
+app.post('/unwrapKey', async (req, res) => {
```

Comment:
fwiw, i think uppercase in path names is weird because the handling of case is inconsistent across the broader "internet toolchain".  i'd suggest using kebab style:

```ts
app.post('/unwrap-key', async (req, res) => {
```

Reply:
d'oh, yes, i agree with this 100%